python_modules/libraries/dagster-aws/dagster_aws/s3/ops.py

```diff
@@ -75,9 +74,7 @@ def file_handle_to_s3(context, file_handle):
 
         yield AssetMaterialization(
             asset_key=s3_file_handle.s3_path,
-            metadata_entries=[
-                MetadataEntry(last_key(key), value=MetadataValue.path(s3_file_handle.s3_path))
-            ],
+            metadata={last_key(key): MetadataValue.path(s3_file_handle.s3_path)},
```

Comment:
I don't think AssetMaterialization has a `metadata` field. Or if it does, and I'm not seeing it, why switch to using `metadata` in this case, but not in the other `AssetMaterialization`s that are getting wrapped in `warnings.catch_warnings()`?

Reply:
> I don't think AssetMaterialization has a metadata field. 

It doesn't have a `metadata` field (yet), but it has a `metadata` argument to its constructor.

> why switch to using metadata in this case, but not in the other AssetMaterializations that are getting wrapped in warnings.catch_warnings()?

Given infinite time, I would have liked to switch to it in all cases. However, I punted on a few tricky ones for now. Those tricky ones are mainly tricky because they merge metadata from multiple sources, and we need to figure what we want to do if the metadata labels conflict.

Reply:
ah cool - missed the `__new__` constructor and was only looking at the namedtuple